lib/unleash/constraint.rb

```diff
@@ -55,9 +55,9 @@ def matches_context?(context)
 
```

Comment:
Extra bike-shedding, and sort of a note to self:

Note that maybe `val1` and `val2` are inverted in relation to who is `context_value` and `constraint_value` in the logging/fallback scenario.

 Maybe we should rename `val1` and `val2` to `context_value` and `constraint_value`?

Here is a suggestion for type annotation, as the function parameters are no longer equivalent. Though maybe that would a good idea? 
```suggestion

    # @param val1 [String]
    # @param val2 [DateTime, String]
```

Reply:
Yeah, completely agree but I think this belongs in a separate PR